front/src/hooks/useUnassignTags.ts

```diff
@@ -0,0 +1,127 @@
+import { useMutation } from "@apollo/client";
+import { graphql } from "../../../graphql/graphql";
+import { useCallback, useState } from "react";
+import { useErrorHandling } from "./useErrorHandling";
+
+export enum IUnassignTagsResultKind {
+  SUCCESS = "success",
+  FAIL = "fail",
+  NETWORK_FAIL = "networkFail",
+  BAD_USER_INPUT = "badUserInput", // no Boxes were pased to the function
+  INVALID_IDENTIFIERS = "invalidIdentifiers",
+}
+
+export interface IUnassignTagsResult {
+  kind: IUnassignTagsResultKind;
+  requestedLabelIdentifiers: string[];
+  requestedTagIds: number[];
+  // TODO: Might not need
+  // successfulLabelIdentifiers?: string[];
+  // invalidIdentifiers?: string[];
+  error?: any;
+}
+
+/**
+ * 
+ * @returns 
+ * {
+  "data": {
+    "unassignTagsFromBoxes": {
+      "invalidBoxLabelIdentifiers": [
+        "123"
+      ],
+      "tagErrorInfo": [],
+      "updatedBoxes": []
+    }
+  }
+}
+ */
+
+export const UNASSIGN_TAGS_FROM_BOXES = graphql(`
+  mutation UnassignTagsFromBoxes($labelIdentifiers: [String!]!, $tagIds: [Int!]!) {
+    unassignTagsFromBoxes(updateInput: { labelIdentifiers: $labelIdentifiers, tagIds: $tagIds }) {
+      updatedBoxes {
+        labelIdentifier
```

Comment:
@jlin95 if you add in the response of the mutation the tag id like this

```
updatedBoxes { 
...
  tag {
    id
  }
...  
}
```

then apollo updates cache of the corresponding entities of type Box and drops their references to tags.

Reply:
`tags { id }` to be correct :wink: magic!